### PR TITLE
Feature/show display feature

### PR DESCRIPTION
**Pull request by stephaniemagdic**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- Year/:year route now will pull by params key of year to fetch data from API at year to map over and create ShowCard components for every show
- The route will display card details for every show in that year (details include date and show key values)
- Each card is a link so that when that card is clicked it will take the user to a the correct show route which will later be completed so that that route will grab tracks for that specific show to render on the page at the unique route.

This closes PR #3 

## What (if anything) did you refactor?
- Imports were fixed
- Some fixes were made to routes including a return and "/" before paths to fix problems with rendering

## Were there any issues that arose?
- NA

## Is there anything that you need from your teammate?
- Please take a look at code added :) 

## Any other comments, questions, or concerns?
- Container component used to map and render each show includes conditional rendering so that in the future, tracks and years can be displayed using the same component. Note... this component is less `reusable` because we have to have to make fetches in these containers in use effect depending on type passed as prop. A way to clean up the container is to create a parent component for each page, otherwise we have to  stick to conditional rendering of some sort if we want to reuse the component. Similarly className is assigned to props.type to help us conditionally style based on data it holds.
- Cypress tests and error page need to be added to complete testing and edge cases. I will create issues.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [X] open index.html
- [X] console.log()
- [X] dev tools

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/container/Container.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import Show from '../show_card/ShowCard'
+import './Container.css'
+
+const Container = (props) => {
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ shows, setShows ] = useState([]);
+
+  //TO DO: includes conditional so that we can use this for our tracks and our years. 
+  useEffect(() => {
+    if (props.type === "shows") {
```
Comment (justincanthony) [inline src/components/container/Container.js:12]: I haven't used proptypes yet, seems pretty straightforward. Looking forward to getting the rundown on them.

File: src/components/container/Container.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      fetchData(`/years/${props.id}`).then(data => {
+        setShows(data.data);
+        setIsLoading(false);
+      }).catch(err => console.log(err))
+      //TO DO: when error component is made, render error component.
```
Comment (justincanthony) [inline src/components/container/Container.js:17]: We can use toastify to render error handling- it may already be in Nina's PR. we can go over it as a group.

File: src/components/container/Container.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ shows, setShows ] = useState([]);
+
+  //TO DO: includes conditional so that we can use this for our tracks and our years. 
+  useEffect(() => {
+    if (props.type === "shows") {
+      fetchData(`/years/${props.id}`).then(data => {
+        setShows(data.data);
+        setIsLoading(false);
+      }).catch(err => console.log(err))
```
Comment (justincanthony) [inline src/components/container/Container.js:16]: We learned in the hooks lesson class yesterday the Resolving promises (.then) cannot be located within the useEffect method.  We will need to declare a function above the useEffect the handles the promise, then call that method in the useEffect

File: src/components/show_card/ShowCard.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { Link } from 'react-router-dom';
+import './ShowCard.css'
+
+const ShowCard = ({show}) => {
+  const {date, venue_name, id} = show;
+  return (
+    <Link to={`/gone-phishin/shows/${id}`} className="link">
+      <article className="ShowCard">
```
Comment (justincanthony) [inline src/components/show_card/ShowCard.js:8]: I may have missed it, but do we need a key here as well?

**Review by justincanthony — CHANGES_REQUESTED** [review]
Awesome Progress! Thank you! 

Please see comments in the code!

File: package.json [hunk 5]
```diff
@@ -36,5 +37,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (ninabrissey) [inline package.json:41]: Thank you for installing. I will add testing in the next couple of days.

File: src/components/app/App.js [hunk 6]
```diff
@@ -1,54 +1,56 @@
...
   return (
-    <ShowContextProvider>
+    <>
+     <ShowContextProvider>
       <Switch>
         <Route
           exact
-          path="gone-phishin/years"
-          render={() => {
+          path="/gone-phishin/years"
```
Comment (ninabrissey) [inline src/components/app/App.js:15]: On my pr. I changed the path to `"/"` We can change it back. I just am not sure how to direct it the above path without giving the years container a <Link>

File: src/components/app/App.js [hunk 7]
```diff
@@ -1,54 +1,56 @@
...
-                <Footer />
+                {/* <Footer /> */}
               </div>
             );
           }}
         />
         <Route render={() => <p>Error. Page Not Found!</p>} />
       </Switch>
     </ShowContextProvider>
+    </>
```
[the remark this answers is not in the document]
Comment (ninabrissey) [reply src/components/app/App.js:53]: You will see this in my PR @stephaniemagdic

File: src/components/container/Container.js (hunk 1 above)
Comment (ninabrissey) [reply src/components/container/Container.js:12]: I don't think this is Proptypes. Looks like she is sending a `type` through props in order to conditionally render. But I am really curious if it is going to be beneficial to a reusable container or if it is going to make the code way less clear.

File: src/components/show_card/ShowCard.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import { Link } from 'react-router-dom';
+import './ShowCard.css'
+
+const ShowCard = ({show}) => {
```
Comment (ninabrissey) [inline src/components/show_card/ShowCard.js:4]: This type of destructuring is something that we have to get into the habit of doing all the time, so great implementation.

**Review by justincanthony — APPROVED** [review]

File: src/components/container/Container.js (hunk 1 above)
Comment (stephaniemagdic) [reply src/components/container/Container.js:12]: Yep, Nina! This is for the conditionally rendering.

**Comment by ninabrissey** [comment]
Nice review team!
